### ISSUE TITLE
sendMessage with certain characters

### ISSUE THREAD

**Issue by Brandantl** [yourWaifu/sleepy-discord#59]
If you use sendMessage with a quote in it the Discord.

```
 			if (message.content == "test")
 			{
 				sendMessage(message.channelID, "Test:");
 				sendMessage(message.channelID, "\"");
 			}
```

> Error 400: The request was improperly formatted, or the server couldn't understand it
> Error 50006: Cannot send an empty message

**Comment by HunterCop224** [comment]
I did a quick test with cout and printing `"\""` prints one `"` to console.

I noticed that if you type `\"` into discord as a user the `\` is removed automatically and the `"` remains. so it could be possible that Discord is parsing the sent string in an unexpected way.

And as a tip and workaround for that, try to minimize the usage of `sendMessage` as you can be rate limited if your not careful. You can use a newline `\n` within the string to put text on the next line and with C++ `\` is escaped with `\` so you'll end up with just "Test:\\n\"".

**Comment by Brandantl** [comment]
I assume the documentation was just for special characters such as \n and not actual normal characters. Is there a list of characters that need to be escaped?

**Comment by yourWaifu** [comment]
Yes, actually this website has a good list of these characters http://json.org/. Most people may need ``\n`` and ``\"`` but this may answer you question.
```
\"
\\
\/
\b
\f
\n
\r
\t
```
